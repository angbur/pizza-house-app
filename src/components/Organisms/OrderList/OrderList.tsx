import PriceTag from 'components/Atoms/PriceTag/PriceTag';
import Typography from 'components/Atoms/Typography/Typography';
import OrderListItem from 'components/Molecules/OrderListItem/OrderListItem';
import { useAppSelector } from 'hooks';
import styled from 'styled-components';
import { OrderItem } from 'types/Order';
import { selectListOfOrderItems, selectSumOfOrder } from '../../Pages/Order/orderSlice';

const StyledList = styled.ul`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 1.5rem;
`;

const StyledDiv = styled.div`
  display: flex;
  gap: 3rem;
  margin: 2rem;
`;

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const OrderList = () => {
  const orderList = useAppSelector(selectListOfOrderItems);
  const sumOfOrder = useAppSelector(selectSumOfOrder);

  return orderList.length === 0 ? (
    <Typography variant='paragraph' color='light'>
      You have not ordered anything yet.
    </Typography>
  ) : (
    <Container>
      <StyledList>
        {orderList.map((el: OrderItem) => (
          <OrderListItem key={`listItem-${el._id}`} item={el} />
        ))}
      </StyledList>
      <StyledDiv>
        <Typography variant='caption' color='light'>
          SUM
        </Typography>
        <PriceTag price={sumOfOrder} size='lg' />
      </StyledDiv>
    </Container>
  );
};

export default OrderList;
