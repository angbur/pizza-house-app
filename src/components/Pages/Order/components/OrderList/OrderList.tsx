import Button from 'components/Atoms/Button/Button';
import PriceTag from 'components/Atoms/PriceTag/PriceTag';
import Typography from 'components/Atoms/Typography/Typography';
import OrderListItem from 'components/Organisms/OrderListItem/OrderListItem';
import { useAppDispatch, useAppSelector } from 'hooks';
import styled from 'styled-components';
import { OrderItem } from 'types/Order';
import { removeAllOrder, selectListOfOrderItems, selectSumOfOrder } from '../../orderSlice';

const StyledList = styled.ul`
  display: flex;
  flex-direction: column;
  gap: 1.5rem;
`;

const StyledDiv = styled.div`
  display: flex;
  gap: 3rem;
  margin: 2rem;
`;

const OrderList = () => {
  const dispatch = useAppDispatch();
  const orderList = useAppSelector(selectListOfOrderItems);
  const sumOfOrder = useAppSelector(selectSumOfOrder);

  const handleRemoveAll = () => {
    dispatch(removeAllOrder());
  };

  return orderList.length === 0 ? (
    <Typography variant='paragraph' color='light'>
      You have not ordered anything yet.
    </Typography>
  ) : (
    <>
      <StyledList>
        {orderList.map((el: OrderItem) => (
          <OrderListItem key={`listItem-${el._id}`} item={el} />
        ))}
      </StyledList>
      <StyledDiv>
        <Typography variant='caption' color='light'>
          SUM
        </Typography>
        <PriceTag price={sumOfOrder} size='lg' />
      </StyledDiv>
      <StyledDiv>
        <Button variant='button-text-light' onClick={handleRemoveAll}>
          Remove all
        </Button>
        <Button variant='primary-light'>Order</Button>
      </StyledDiv>
    </>
  );
};

export default OrderList;
