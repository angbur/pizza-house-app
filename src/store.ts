import { configureStore } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import { pizzaApi } from 'services/pizza';

export const store = configureStore({
  reducer: {
    order: orderReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(pizzaApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
