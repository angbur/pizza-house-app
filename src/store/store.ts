import { configureStore, Store } from '@reduxjs/toolkit';
import orderReducer from 'components/Pages/Order/orderSlice';
import dialogReducer from 'components/Organisms/Dialog/dialogSlice';
import { pizzaApi } from 'services/pizza';
import orderMiddleware from './middleware/orderMiddleware';

export const store: Store = configureStore({
  reducer: {
    order: orderReducer,
    dialog: dialogReducer,
    [pizzaApi.reducerPath]: pizzaApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(pizzaApi.middleware, orderMiddleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
