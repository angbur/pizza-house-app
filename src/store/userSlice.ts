import { createSlice } from '@reduxjs/toolkit';
import { RootState } from './store';
import { toast } from 'react-toastify';
import { userApi } from 'services/user';

type User = {
  loggedIn: boolean;
  status: string | undefined;
  error: string | undefined;
  userId: string | undefined;
  login: string | undefined;
  token: string | undefined;
};

const initialState: User = {
  loggedIn: false,
  status: 'idle',
  error: undefined,
  userId: undefined,
  login: undefined,
  token: undefined,
};

const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    logout: (state) => {
      state.loggedIn = false;
      state.userId = undefined;
      state.login = undefined;
      state.token = undefined;
      toast.success('Logged out!');
    },
  },
  extraReducers: (builder) => {
    builder.addMatcher(userApi.endpoints.login.matchFulfilled, (state, { payload }) => {
      state.loggedIn = true;
      state.token = payload.token;
      state.userId = payload.userId;
      state.login = payload.login;
    });
  },
});

export const { logout } = userSlice.actions;

export const selectName = (state: RootState) => state.user.user;

export const isLoggedIn = (state: RootState) => state.user.loggedIn;

export const selectResponseStatus = (state: RootState) => state.user.status;

export const selectResponseError = (state: RootState) => state.user.error;

export const selectLoggedInUser = (state: RootState) => state.user.user;

export default userSlice.reducer;
