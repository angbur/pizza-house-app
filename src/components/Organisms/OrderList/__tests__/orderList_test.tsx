import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import { store } from 'store/store';
import { Provider } from 'react-redux';
import OrderList from '../OrderList';

const Wrapper = () => {
  return (
    <ThemeProvider>
      <Provider store={store}>
        <OrderList />
      </Provider>
    </ThemeProvider>
  );
};

describe('Order List', () => {
  it('should renders order list correctly', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
