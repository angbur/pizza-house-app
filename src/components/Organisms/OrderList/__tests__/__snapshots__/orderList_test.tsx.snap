// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order List should renders order list correctly 1`] = `
<p
  className="paragraph"
  style={
    Object {
      "color": "#FCFBFB",
      "fontFamily": "Source Sans Pro",
      "fontSize": "1rem",
      "fontWeight": "400",
      "letterSpacing": "0",
      "lineHeight": "1.6",
    }
  }
>
  You have not ordered anything yet.
</p>
`;
